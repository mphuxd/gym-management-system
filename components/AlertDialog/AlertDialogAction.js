import React from "react";
import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";

const AlertDialogAction = React.forwardRef(({ children, ...props }, forwardedRef) => {
  return (
    <AlertDialogPrimitive.Action ref={forwardedRef} {...props}>
      {children}
    </AlertDialogPrimitive.Action>
  );
});

AlertDialogAction.displayName = "AlertDialogAction";

export default AlertDialogAction;
