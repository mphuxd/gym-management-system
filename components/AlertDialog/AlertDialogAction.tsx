import React from 'react';
import * as AlertDialogPrimitive from '@radix-ui/react-alert-dialog';

type AlertDialogActionRef = React.ElementRef<
  typeof AlertDialogPrimitive.Action
>;

export interface AlertDialogActionProps
  extends AlertDialogPrimitive.AlertDialogActionProps {
  children?: React.ReactNode;
}

const AlertDialogAction = React.forwardRef<
  AlertDialogActionRef,
  AlertDialogActionProps
>(({ children, ...props }, forwardedRef) => (
  <AlertDialogPrimitive.Action ref={forwardedRef} {...props}>
    {children}
  </AlertDialogPrimitive.Action>
));

AlertDialogAction.displayName = 'AlertDialogAction';

export default AlertDialogAction;
