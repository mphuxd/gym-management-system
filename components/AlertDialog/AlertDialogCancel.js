import React from "react";
import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";

const AlertDialogCancel = React.forwardRef(({ children, ...props }, forwardedRef) => {
  return (
    <AlertDialogPrimitive.Cancel ref={forwardedRef} {...props}>
      {children}
    </AlertDialogPrimitive.Cancel>
  );
});

AlertDialogCancel.displayName = "AlertDialog";

export default AlertDialogCancel;
