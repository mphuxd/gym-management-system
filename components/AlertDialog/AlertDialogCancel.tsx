import React from 'react';
import * as AlertDialogPrimitive from '@radix-ui/react-alert-dialog';

type AlertDialogCancelRef = React.ElementRef<
  typeof AlertDialogPrimitive.Cancel
>;

export interface AlertDialogCancelProps
  extends AlertDialogPrimitive.AlertDialogCancelProps {
  children?: React.ReactNode;
}

const AlertDialogCancel = React.forwardRef<
  AlertDialogCancelRef,
  AlertDialogCancelProps
>(({ children, ...props }, forwardedRef) => (
  <AlertDialogPrimitive.Cancel ref={forwardedRef} {...props}>
    {children}
  </AlertDialogPrimitive.Cancel>
));

AlertDialogCancel.displayName = 'AlertDialog';

export default AlertDialogCancel;
