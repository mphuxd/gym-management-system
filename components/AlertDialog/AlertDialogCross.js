import React from 'react';
import { Cross2Icon } from '@radix-ui/react-icons';

function AlertDialogCross(props) {
  return (
    <button
      {...props}
      type="button"
      className="bg-transparent p-2 hover:bg-neg-hover hover:text-white active:bg-neg-active active:outline-none"
    >
      <Cross2Icon className="fill-icon" />
    </button>
  );
}

export default AlertDialogCross;
