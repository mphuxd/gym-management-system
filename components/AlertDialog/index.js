// eslint-disable-next-line import/no-cycle
export { default as AlertDialog } from './AlertDialog';
export { default as AlertDialogAction } from './AlertDialogAction';
export { default as AlertDialogCancel } from './AlertDialogCancel';
