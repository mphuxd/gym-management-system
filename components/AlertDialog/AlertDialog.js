import React, { useState } from 'react';
import * as AlertDialogPrimitive from '@radix-ui/react-alert-dialog';
import { Cross2Icon } from '@radix-ui/react-icons';
import { ErrorFilled } from '@carbon/icons-react';
import { useAtom } from 'jotai';
import { Button } from '@/components';
import { toastAtom } from '@/atoms';

const AlertDialog = React.forwardRef(
  (
    {
      children,
      title,
      description,
      close,
      action,
      isOpen,
      setIsOpen,
      href,
      intent,
      actionPhrase = 'delete me',
      toastTitle,
      toastDescription,
      ...props
    },
    forwardedRef
  ) => {
    const [input, setInput] = useState();
    const [error, setError] = useState();
    // eslint-disable-next-line no-unused-vars
    const [toast, setToast] = useAtom(toastAtom);

    async function handleAction() {
      const response = await fetch(href);
      const results = await response.json();
      if (results.statusCode === 200) {
        setIsOpen(false);
        if (toastTitle && toastDescription) {
          setToast({
            title: toastTitle,
            description: toastDescription,
            isOpen: true,
            intent: 'success',
          });
        }
      }
      if (results.statusCode === 403) {
        setError(results.message);
        setIsOpen(true);
      }
    }

    const handleOpenChange = () => {
      setIsOpen();
      setError('');
      setInput('');
    };

    return (
      <AlertDialogPrimitive.Root
        open={isOpen}
        onOpenChange={handleOpenChange}
        ref={forwardedRef}
        {...props}
      >
        <AlertDialogPrimitive.Portal>
          <AlertDialogPrimitive.Overlay className="fixed bg-mauve12 opacity-70 inset-0" />
          <AlertDialogPrimitive.Content
            onClick={(e) => {
              e.preventDefault();
              e.stopPropagation();
            }}
            className="absolute inset-1/2 -translate-x-1/2 -translate-y-3/4 h-fit w-[640px] bg-white p-6 rounded-lg space-y-4"
          >
            <div className="flex flex-row justify-between">
              <AlertDialogPrimitive.Title className="font-semibold">
                {title}
              </AlertDialogPrimitive.Title>
              <div className="p-2 block hover:bg-slate4 rounded-sm focus:outline focus:outline-slate7 active:bg-slate5">
                <Cross2Icon
                  onClick={handleOpenChange}
                  className="hover:cursor-pointer"
                />
              </div>
            </div>
            <AlertDialogPrimitive.Description>
              {description}
            </AlertDialogPrimitive.Description>
            {intent === 'constrained' && (
              <div className="flex flex-col">
                <div className="mt-4 text-gray11">
                  To continue, enter the phrase
                  <span className="font-bold text-black">{` ${actionPhrase}`}</span>
                </div>
                {error && (
                  <div className="flex flex-row items-center mt-2 gap-x-1 ">
                    <ErrorFilled className="fill-red11" />
                    <div className="text-red11">{error}</div>
                  </div>
                )}
                <input
                  placeholder={actionPhrase}
                  type="text"
                  input={input}
                  onInput={(e) => setInput(e.target.value)}
                  className="w-full outline outline-1 outline-gray10 px-2 py-1 my-4"
                />
              </div>
            )}
            <div className="flex flex-row justify-end gap-x-6">
              <AlertDialogPrimitive.Cancel
                asChild
                onClick={(e) => e.stopPropagation()}
              >
                <Button as="button">{close}</Button>
              </AlertDialogPrimitive.Cancel>
              <AlertDialogPrimitive.Action
                asChild
                onClick={(e) => {
                  e.stopPropagation();
                  e.preventDefault();
                  handleAction(e);
                }}
              >
                {intent === 'constrained' ? (
                  <Button
                    disabled={input !== actionPhrase}
                    intent={input !== actionPhrase ? 'disabled' : 'danger'}
                    as="button"
                  >
                    {action}
                  </Button>
                ) : (
                  <Button intent="danger" as="button">
                    {action}
                  </Button>
                )}
              </AlertDialogPrimitive.Action>
            </div>
            {children}
          </AlertDialogPrimitive.Content>
        </AlertDialogPrimitive.Portal>
      </AlertDialogPrimitive.Root>
    );
  }
);

AlertDialog.displayName = 'AlertDialog';

export default AlertDialog;
