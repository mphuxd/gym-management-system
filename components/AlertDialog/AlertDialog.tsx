import React, { useState } from 'react';
import * as AlertDialogPrimitive from '@radix-ui/react-alert-dialog';
import { ErrorFilled } from '@carbon/icons-react';
import { useAtom } from 'jotai';
import { AlertDialogCross, Button } from '@/components';
import { toastAtom } from '@/atoms';

export interface AlertDialogProps
  extends AlertDialogPrimitive.AlertDialogProps {
  children?: React.ReactNode;
  title?: string;
  description?: string;
  close?: string;
  action?: string;
  isOpen?: boolean;
  setIsOpen?: React.Dispatch<React.SetStateAction<boolean>>;
  href?: string;
  intent?: string;
  actionPhrase?: string;
  toastTitle?: string;
  toastDescription?: string;
}

function AlertDialog({
  action,
  actionPhrase = 'delete me',
  children,
  close,
  description,
  href,
  intent,
  isOpen,
  setIsOpen,
  title,
  toastDescription,
  toastTitle,
  ...props
}) {
  const [input, setInput] = useState<string | null>();
  const [error, setError] = useState<string | null>();
  const [, setToast] = useAtom(toastAtom);

  function sendToast() {
    if (toastTitle && toastDescription) {
      setToast({
        title: toastTitle,
        description: toastDescription,
        isOpen: true,
        intent: 'success',
      });
    }
  }

  async function handleAction() {
    const response = await fetch(href);
    const results = await response.json();
    if (results.statusCode === 200) {
      setIsOpen(false);
      sendToast();
    }
    if (results.statusCode === 403) {
      setError(results.message);
      setIsOpen(true);
    }
  }

  const handleOpenChange = (e) => {
    if (e) {
      e.stopPropagation();
    }
    setIsOpen(!isOpen);
    setError('');
    setInput('');
  };

  return (
    <AlertDialogPrimitive.Root
      {...props}
      open={isOpen}
      onOpenChange={handleOpenChange}
    >
      <AlertDialogPrimitive.Portal>
        <AlertDialogPrimitive.Overlay
          onClick={handleOpenChange}
          className="fixed inset-0 z-10 bg-overlay-neg opacity-70"
        />
        <AlertDialogPrimitive.Content
          tabIndex={-1}
          onClick={(e) => {
            e.preventDefault();
            e.stopPropagation();
          }}
          className="absolute inset-1/2 z-20 h-fit w-[640px] -translate-x-1/2 -translate-y-3/4 space-y-4 bg-layer p-6"
        >
          <div className="flex flex-row justify-between">
            <AlertDialogPrimitive.Title className="font-semibold">
              {title}
            </AlertDialogPrimitive.Title>
            <AlertDialogCross onClick={handleOpenChange} />
          </div>
          <AlertDialogPrimitive.Description asChild>
            <div>
              {description}
              {intent === 'constrained' && (
                <div className="flex flex-col">
                  <div>
                    <span className="mt-4 inline-block text-support">
                      To continue, enter the phrase
                    </span>
                    <span className="font-bold">{` ${actionPhrase}`}</span>
                  </div>
                  {error && (
                    <div className="mt-2 flex flex-row items-center gap-x-1 text-neg-text">
                      <ErrorFilled className="fill-icon-neg" />
                      <div>{error}</div>
                    </div>
                  )}
                  <input
                    placeholder={actionPhrase}
                    type="text"
                    onInput={(e) =>
                      setInput((e.target as HTMLInputElement).value)
                    }
                    className="my-4 w-full px-2 py-1 outline outline-1 outline-border-interactive"
                  />
                </div>
              )}
            </div>
          </AlertDialogPrimitive.Description>

          <div className="flex flex-row justify-end gap-x-6">
            <AlertDialogPrimitive.Cancel
              asChild
              onClick={(e) => e.stopPropagation()}
            >
              <Button as="button" role="button" intent="ghost" size="large">
                {close}
              </Button>
            </AlertDialogPrimitive.Cancel>
            <AlertDialogPrimitive.Action
              asChild
              onClick={(e) => {
                e.stopPropagation();
                e.preventDefault();
                handleAction();
              }}
            >
              {intent === 'constrained' ? (
                <Button
                  as="button"
                  role="button"
                  disabled={input !== actionPhrase}
                  intent={input !== actionPhrase ? 'disabled' : 'danger'}
                >
                  {action}
                </Button>
              ) : (
                <Button intent="danger" as="button">
                  {action}
                </Button>
              )}
            </AlertDialogPrimitive.Action>
          </div>
          {children}
        </AlertDialogPrimitive.Content>
      </AlertDialogPrimitive.Portal>
    </AlertDialogPrimitive.Root>
  );
}

AlertDialog.displayName = 'AlertDialog';

export default AlertDialog;
