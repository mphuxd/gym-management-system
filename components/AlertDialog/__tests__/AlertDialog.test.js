import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import AlertDialog from '../AlertDialog';

describe('AlertDialog ', () => {
  it('should render "default" intent', async () => {
    // eslint-disable-next-line no-unused-vars
    render(
      <AlertDialog
        isOpen
        title="Test Alert Dialog title"
        description="Test Alert Dialog description"
        close="Alert Dialog Close"
        action="Alert Dialog Action"
        href="/"
      />
    );
    const alertDialog = await screen.findByRole('alertdialog');
    expect(alertDialog).toBeInTheDocument();
    expect(alertDialog).toMatchSnapshot();
  });

  it('should render "constrained" intent', async () => {
    render(
      <AlertDialog
        isOpen
        intent="constrained"
        actionPhrase="test action phrase"
        title="Test Alert Dialog title"
        description="Test Alert Dialog description"
        close="Alert Dialog Close"
        action="Alert Dialog Action"
        href="/"
      />
    );
    const alertDialog = await screen.findByRole('alertdialog');
    expect(alertDialog).toBeInTheDocument();
    expect(alertDialog).toMatchSnapshot();

    const input = await screen.findByPlaceholderText('test action phrase');
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  it(' in "constrained" intent, action button should be disabled by default', async () => {
    render(
      <AlertDialog
        isOpen
        intent="constrained"
        actionPhrase="test action phrase"
        title="Test Alert Dialog title"
        description="Test Alert Dialog description"
        close="Alert Dialog Close"
        action="Alert Dialog Action"
        href="/"
      />
    );
    const alertDialog = await screen.findByRole('alertdialog');
    expect(alertDialog).toBeInTheDocument();
    expect(alertDialog).toMatchSnapshot();

    const actionButton = await screen.findByText('Alert Dialog Action');
    expect(actionButton).toBeInTheDocument();
    expect(actionButton).toBeDisabled();
    expect(actionButton).toMatchSnapshot();
  });

  it(' in "constrained" intent, action button should be enabled when input value matches action text', async () => {
    render(
      <AlertDialog
        isOpen
        intent="constrained"
        actionPhrase="test action phrase"
        title="Test Alert Dialog title"
        description="Test Alert Dialog description"
        close="Alert Dialog Close"
        action="Alert Dialog Action"
        href="/"
      />
    );

    const input = await screen.findByPlaceholderText('test action phrase');
    expect(input).toBeInTheDocument();
    await userEvent.type(input, 'test action phrase');

    const actionButton = await screen.findByText('Alert Dialog Action');
    expect(actionButton).not.toBeDisabled();
    expect(actionButton).toMatchSnapshot();

    const alertDialog = await screen.findByRole('alertdialog');
    expect(alertDialog).toMatchSnapshot();
  });
});
