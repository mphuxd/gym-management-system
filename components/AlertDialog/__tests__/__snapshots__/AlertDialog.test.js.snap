// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog   in "constrained" intent, action button should be disabled by default 1`] = `
<div
  aria-describedby="radix-:r8:"
  aria-labelledby="radix-:r7:"
  class="absolute inset-1/2 -translate-x-1/2 -translate-y-3/4 h-fit w-[640px] bg-white p-6 rounded-lg space-y-4"
  data-state="open"
  id="radix-:r6:"
  role="alertdialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div
    class="flex flex-row justify-between"
  >
    <h2
      class="font-semibold"
      id="radix-:r7:"
    >
      Test Alert Dialog title
    </h2>
    <div
      class="p-2 block hover:bg-slate4 rounded-sm focus:outline focus:outline-slate7 active:bg-slate5"
    >
      <svg
        class="hover:cursor-pointer"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
  <p
    id="radix-:r8:"
  >
    Test Alert Dialog description
  </p>
  <div
    class="flex flex-col"
  >
    <div
      class="mt-4 text-gray11"
    >
      To continue, enter the phrase
      <span
        class="font-bold text-black"
      >
         test action phrase
      </span>
    </div>
    <input
      class="w-full outline outline-1 outline-gray10 px-2 py-1 my-4"
      placeholder="test action phrase"
      type="text"
    />
  </div>
  <div
    class="flex flex-row justify-end gap-x-6"
  >
    <button
      class="focus:outline text-slate11 border-slate9 border-[1px] hover:text-slate12 hover:shadow-sm hover:shadow-sand8 hover:bg-slate3 hover:border-slate10 hover:ring-0 focus:bg-slate4 active:bg-slate5 px-4 py-1 w-fit rounded-md"
      type="button"
    >
      Alert Dialog Close
    </button>
    <button
      class="focus:outline text-gray8 outline-gray6 outline outline-2 px-4 py-1 w-fit rounded-md"
      disabled=""
      type="button"
    >
      Alert Dialog Action
    </button>
  </div>
</div>
`;

exports[`AlertDialog   in "constrained" intent, action button should be disabled by default 2`] = `
<button
  class="focus:outline text-gray8 outline-gray6 outline outline-2 px-4 py-1 w-fit rounded-md"
  disabled=""
  type="button"
>
  Alert Dialog Action
</button>
`;

exports[`AlertDialog   in "constrained" intent, action button should be enabled when input value matches action text 1`] = `
<button
  class="focus:outline text-red11 bg-red4 hover:bg-red5 active:bg-red6  focus:outline-red7 px-4 py-1 w-fit rounded-md"
  type="button"
>
  Alert Dialog Action
</button>
`;

exports[`AlertDialog   in "constrained" intent, action button should be enabled when input value matches action text 2`] = `
<div
  aria-describedby="radix-:rb:"
  aria-labelledby="radix-:ra:"
  class="absolute inset-1/2 -translate-x-1/2 -translate-y-3/4 h-fit w-[640px] bg-white p-6 rounded-lg space-y-4"
  data-state="open"
  id="radix-:r9:"
  role="alertdialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div
    class="flex flex-row justify-between"
  >
    <h2
      class="font-semibold"
      id="radix-:ra:"
    >
      Test Alert Dialog title
    </h2>
    <div
      class="p-2 block hover:bg-slate4 rounded-sm focus:outline focus:outline-slate7 active:bg-slate5"
    >
      <svg
        class="hover:cursor-pointer"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
  <p
    id="radix-:rb:"
  >
    Test Alert Dialog description
  </p>
  <div
    class="flex flex-col"
  >
    <div
      class="mt-4 text-gray11"
    >
      To continue, enter the phrase
      <span
        class="font-bold text-black"
      >
         test action phrase
      </span>
    </div>
    <input
      class="w-full outline outline-1 outline-gray10 px-2 py-1 my-4"
      input="test action phrase"
      placeholder="test action phrase"
      type="text"
    />
  </div>
  <div
    class="flex flex-row justify-end gap-x-6"
  >
    <button
      class="focus:outline text-slate11 border-slate9 border-[1px] hover:text-slate12 hover:shadow-sm hover:shadow-sand8 hover:bg-slate3 hover:border-slate10 hover:ring-0 focus:bg-slate4 active:bg-slate5 px-4 py-1 w-fit rounded-md"
      type="button"
    >
      Alert Dialog Close
    </button>
    <button
      class="focus:outline text-red11 bg-red4 hover:bg-red5 active:bg-red6  focus:outline-red7 px-4 py-1 w-fit rounded-md"
      type="button"
    >
      Alert Dialog Action
    </button>
  </div>
</div>
`;

exports[`AlertDialog  should render "constrained" intent 1`] = `
<div
  aria-describedby="radix-:r5:"
  aria-labelledby="radix-:r4:"
  class="absolute inset-1/2 -translate-x-1/2 -translate-y-3/4 h-fit w-[640px] bg-white p-6 rounded-lg space-y-4"
  data-state="open"
  id="radix-:r3:"
  role="alertdialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div
    class="flex flex-row justify-between"
  >
    <h2
      class="font-semibold"
      id="radix-:r4:"
    >
      Test Alert Dialog title
    </h2>
    <div
      class="p-2 block hover:bg-slate4 rounded-sm focus:outline focus:outline-slate7 active:bg-slate5"
    >
      <svg
        class="hover:cursor-pointer"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
  <p
    id="radix-:r5:"
  >
    Test Alert Dialog description
  </p>
  <div
    class="flex flex-col"
  >
    <div
      class="mt-4 text-gray11"
    >
      To continue, enter the phrase
      <span
        class="font-bold text-black"
      >
         test action phrase
      </span>
    </div>
    <input
      class="w-full outline outline-1 outline-gray10 px-2 py-1 my-4"
      placeholder="test action phrase"
      type="text"
    />
  </div>
  <div
    class="flex flex-row justify-end gap-x-6"
  >
    <button
      class="focus:outline text-slate11 border-slate9 border-[1px] hover:text-slate12 hover:shadow-sm hover:shadow-sand8 hover:bg-slate3 hover:border-slate10 hover:ring-0 focus:bg-slate4 active:bg-slate5 px-4 py-1 w-fit rounded-md"
      type="button"
    >
      Alert Dialog Close
    </button>
    <button
      class="focus:outline text-gray8 outline-gray6 outline outline-2 px-4 py-1 w-fit rounded-md"
      disabled=""
      type="button"
    >
      Alert Dialog Action
    </button>
  </div>
</div>
`;

exports[`AlertDialog  should render "constrained" intent 2`] = `
<input
  class="w-full outline outline-1 outline-gray10 px-2 py-1 my-4"
  placeholder="test action phrase"
  type="text"
/>
`;

exports[`AlertDialog  should render "default" intent 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="absolute inset-1/2 -translate-x-1/2 -translate-y-3/4 h-fit w-[640px] bg-white p-6 rounded-lg space-y-4"
  data-state="open"
  id="radix-:r0:"
  role="alertdialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div
    class="flex flex-row justify-between"
  >
    <h2
      class="font-semibold"
      id="radix-:r1:"
    >
      Test Alert Dialog title
    </h2>
    <div
      class="p-2 block hover:bg-slate4 rounded-sm focus:outline focus:outline-slate7 active:bg-slate5"
    >
      <svg
        class="hover:cursor-pointer"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
  <p
    id="radix-:r2:"
  >
    Test Alert Dialog description
  </p>
  <div
    class="flex flex-row justify-end gap-x-6"
  >
    <button
      class="focus:outline text-slate11 border-slate9 border-[1px] hover:text-slate12 hover:shadow-sm hover:shadow-sand8 hover:bg-slate3 hover:border-slate10 hover:ring-0 focus:bg-slate4 active:bg-slate5 px-4 py-1 w-fit rounded-md"
      type="button"
    >
      Alert Dialog Close
    </button>
    <button
      class="focus:outline text-red11 bg-red4 hover:bg-red5 active:bg-red6  focus:outline-red7 px-4 py-1 w-fit rounded-md"
      type="button"
    >
      Alert Dialog Action
    </button>
  </div>
</div>
`;
